gapis/api/vulkan/api/pipeline.api

```diff
@@ -305,24 +312,31 @@ cmd VkResult vkCreateGraphicsPipelines(
         spec_data.Data = clone(as!u8*(spec_info.pData)[0:spec_info.dataSize])
         stage_data.Specialization = spec_data
       }
-      
+
       if stage_data.Module.Descriptors != null {
         moduleDescriptors := stage_data.Module.Descriptors.Descriptors[as!string(stage_create_info.pName)]
         for _, _, d in moduleDescriptors {
-          obj.UsedDescriptors[len(obj.UsedDescriptors)] = d
-        }
+          if !seenDescs.Set[(as!u64(d.Set)<<32) | as!u64(d.Binding)] {
+            obj.UsedDescriptors[len(obj.UsedDescriptors)] = d
+            seenDescs.Set[(as!u64(d.Set)<<32) | as!u64(d.Binding)] = true
+          }
+        } 
       } else {
         nSets := len(obj.Layout.SetLayouts)
```

Comment:
What is this second branch even for?

Reply:
@AWoloszyn this looks like "consider all descriptors to be used if we couldn't figure it out" -- is this a path we actually want to take, or should we just treat this as a bug?

Reply:
Used descriptors is an optimisation, so if we can't figure it out, then falling back to slow path is reasonable to me.

Also making the tool useless to our users if we can't parse a shader seems unhelpful, when we could provide the same functionality in a different way.

Secondly since this is an external project we use, getting fixes may take longer than we would like.